packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -25,14 +25,28 @@ import {ZERO_HASH} from "../../../constants";
 import {IBeaconDb} from "../../../db";
 import {IEth1ForBlockProduction} from "../../../eth1";
 import {INetwork} from "../../../network";
-import {IBeaconSync} from "../../../sync";
+import {IBeaconSync, SyncMode} from "../../../sync";
 import {toGraffitiBuffer} from "../../../util/graffiti";
 import {IApiOptions} from "../../options";
 import {ApiError} from "../errors";
 import {ApiNamespace, IApiModules} from "../interface";
-import {checkSyncStatus} from "../utils";
 import {IValidatorApi} from "./interface";
 
+/**
+ * Validator clock may be advanced from beacon's clock. If the validator requests a resource in a
+ * future slot, wait some time instead of rejecting the request because it's in the future
+ */
+const MAX_API_CLOCK_DISPARITY_MS = 1000;
```

Comment:
where does the value for this and `SYNC_TOLERANCE_EPOCHS` below come from?

Reply:
SYNC_TOLERANCE_EPOCHS comes from Lighthouse and this 1000 come from my own observations of our metrics, it's a bit arbitrary. It's double the gossip clock disparity to account for our overloaded node.